Mathlib/Order/Interval/Finset/Basic.lean

```diff
@@ -308,6 +308,19 @@ section LocallyFiniteOrderTop
 
 variable [LocallyFiniteOrderTop α]
 
+@[simp]
+theorem Ioi_eq_empty_iff : Ioi a = ∅ ↔ IsMax a := by
+  rw [← coe_eq_empty, coe_Ioi, Set.Ioi_eq_empty_iff]
+
+alias ⟨_, Ioi_eq_empty⟩ := Ioi_eq_empty_iff
+
+@[simp]
+theorem Ioi_top [OrderTop α] : Ioi (⊤ : α) = ∅ := Ioi_eq_empty isMax_top
```

Comment:
Isn't `OrderTop` already implied by the `variable`s above?

Reply:
It is a bit strange, but from the doc, `LocallyFiniteOrderTop` is a mix-in that allows for an empty type, whereas `OrderTop` is not (requires a top element).